rgw/v2/tests/s3cmd/test_large_object_gc.py

```diff
@@ -0,0 +1,173 @@
+from v2.utils.test_desc import AddTestInfo
+from v2.utils.utils import RGWService
+
+log = logging.getLogger()
+
+
+def test_exec(config):
+    """
+    Executes test based on configuration passed
+    Args:
+        config(object): Test configuration
+    """
+    io_info_initialize = IOInfoInitialize()
+    basic_io_structure = BasicIOInfoStructure()
+    io_info_initialize.initialize(basic_io_structure.initial())
+    umgmt = UserMgmt()
+    ceph_conf = CephConfOp()
+    rgw_service = RGWService()
+    # preparing data
+    user_name = resource_op.create_users(no_of_users_to_create=1)[0]["user_id"]
+    tenant = "tenant"
+    tenant_user_info = umgmt.create_tenant_user(
+        tenant_name=tenant, user_id=user_name, displayname=user_name
+    )
+    umgmt.create_subuser(tenant_name=tenant, user_id=user_name)
+
+    ip_and_port = s3cmd_reusable.get_rgw_ip_and_port()
+    s3_auth.do_auth(tenant_user_info, ip_and_port)
+
+    bucket_name = utils.gen_bucket_name_from_userid(user_name, rand_no=0)
+
+    # Create a bucket
+    s3cmd_reusable.create_bucket(bucket_name)
+    log.info(f"Bucket {bucket_name} created")
+
+    # Upload a 2GB file to bucket
+    uploaded_file_info = s3cmd_reusable.upload_file(bucket_name, file_size=2147483648)
+    uploaded_file = uploaded_file_info["name"]
+    uploaded_file_md5 = uploaded_file_info["md5"]
+    log.info(f"Uploaded file {uploaded_file} to bucket {bucket_name}")
+
+    if config.gc_verification is True:
+        log.info("making changes to ceph.conf")
+        config.rgw_gc_obj_min_wait = 5
+        ceph_conf.set_to_ceph_conf(
+            "global",
+            ConfigOpts.rgw_gc_obj_min_wait,
+            str(config.rgw_gc_obj_min_wait),
+        )
+        log.info("trying to restart services")
+        srv_restarted = rgw_service.restart()
+        time.sleep(30)
+        if srv_restarted is False:
+            raise TestExecError("RGW service restart failed")
+        else:
+            log.info("RGW service restarted")
+        log.info("download large object again to make gc list with shadow entries")
+
+        s3cmd_reusable.download_file(
+            bucket_name, uploaded_file, local_file_name="download1.img"
+        )
+        time.sleep(5)
+        downloaded_file1_md5 = utils.get_md5("download1.img")
+        assert uploaded_file_md5 == downloaded_file1_md5
+        gc_list_output = json.loads(
+            utils.exec_shell_cmd("radosgw-admin gc list --include-all")
+        )
+
+        log.info(gc_list_output)
+
+        if gc_list_output:
+            log.info("Shadow obj found after setting rgw_gc_obj_min_wait to 5 sec")
+            utils.exec_shell_cmd("radosgw-admin gc process --include-all")
+            log.info("Object download should not error out in 404 NoSuchKey error")
+            s3cmd_reusable.download_file(
+                bucket_name, uploaded_file, local_file_name="download2.img"
+            )
+            downloaded_file2_md5 = utils.get_md5("download2.img")
```

Comment:
Avoid leaving residues of the test run... or ensure we have a cleanup even if there is an assertion-error

Reply:
Same as above. Will add cleanup script or will call it if it is available already